ignite/contrib/metrics/ExpectedCalibrationError.py

```diff
@@ -0,0 +1,57 @@
+import torch
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics import Metric
+
+
+class ExpectedCalibrationError(Metric):
+    def __init__(self, num_bins=10, device=None):
+        super(ExpectedCalibrationError, self).__init__()
+        self.num_bins = num_bins
+        self.device = device
+        self.reset()
+
+    def reset(self):
+        self.confidences = torch.tensor([], device=self.device)
+        self.corrects = torch.tensor([], device=self.device)
+
+    def update(self, output):
+        y_pred, y = output
+
+        assert y_pred.dim() == 2 and y_pred.shape[1] == 2, "This metric is for binary classification."
```

Comment:
Let's use the following way to raise errors instead of `assert`:
```python
if not (y_pred.dim() == 2 and y_pred.shape[1] == 2):
    raise ValueError("This metric is for binary classification")
```

To assert if the input is binary we were doing previously something like here:
https://github.com/pytorch/ignite/blob/4dc4e04b901c682e3e6f0b193a18c50c56eb3c41/ignite/metrics/accuracy.py#L51-L67